apps/server/src/departments/departments.controller.spec.ts

```diff
@@ -1,20 +1,164 @@
+          name: 'department-name',
+          createdAt: new Date(1672531200),
+          updatedAt: new Date(1672531200),
+        },
+        {
+          id: 'department-id-one',
+          name: 'department-name-one',
+          createdAt: new Date(1672531201),
+          updatedAt: new Date(1672531201),
+        },
+      ];
+
+      const expected = [
+        new DepartmentEntity({
+          id: 'department-id',
+          name: 'department-name',
+          createdAt: new Date(1672531200),
+          updatedAt: new Date(1672531200),
+        }),
+        new DepartmentEntity({
+          id: 'department-id-one',
+          name: 'department-name-one',
+          createdAt: new Date(1672531201),
+          updatedAt: new Date(1672531201),
+        }),
+      ];
+
+      jest
+        .spyOn(departmentsService, 'findAll')
+        .mockImplementation(async () => result);
+
+      expect(await controller.findAll(10)).toEqual(expected);
+    });
+  });
+
+  describe('create', () => {
+    it('should create a new department', async () => {
+      const createDepartmentDto: CreateDepartmentDto = {
+        name: 'New Department',
+        signatureFields: [],
+      };
+
+      const createdDepartment = {
+        id: 'new-department-id',
+        name: 'New Department',
+        createdAt: new Date(),
+        updatedAt: new Date(),
+      };
+
+      jest
+        .spyOn(departmentsService, 'create')
+        .mockImplementation(async () => createdDepartment);
+
+      const result = await controller.create(createDepartmentDto);
+      const expected = new DepartmentEntity(createdDepartment);
+      expect(result).toEqual(expected);
+    });
+  });
+
+  describe('findOne', () => {
+    it('should find a department by ID', async () => {
+      const departmentId = 'valid-department-id';
+      const department = {
+        id: departmentId,
+        name: 'Department Name',
+        createdAt: new Date(),
+        updatedAt: new Date(),
+      };
+
+      jest
+        .spyOn(departmentsService, 'findOne')
+        .mockImplementation(async (id) => {
+          if (id === departmentId) {
+            return department as any;
+          }
+          return null as any;
+        });
+
+      const result = await controller.findOne(departmentId);
+      const expected = new DepartmentEntity(department);
+      expect(result).toEqual(expected);
+    });
+
+    it('should raise a NotFoundException if department is not found', async () => {
+      const invalidDepartmentId = 'invalid-department-id';
+
+      jest
+        .spyOn(departmentsService, 'findOne')
+        .mockImplementation(async (id) => {
+          if (id === invalidDepartmentId) {
+            return null as any; // Cast to the expected type.
+          }
+          // Handle other IDs if needed.
+        });
+
+      await expect(
+        controller.findOne(invalidDepartmentId),
+      ).rejects.toThrowError(NotFoundException);
+    });
+  });
+
+  describe('update', () => {});
```

Comment:
tests tbd
https://trello.com/c/pMztE6Dd/50-add-controller-test-for-updating-department